src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1873,4 +1881,104 @@ private boolean expectedFunction(@Nullable PsiElement resolve) {
       return true;
     }
   }
+
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> NAMES;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      NAMES = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom atom = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      ErlangMaxExpression expr = ObjectUtils.tryCast(atom != null ? atom.getParent() : null, ErlangMaxExpression.class);
+      ErlangArgumentList args = ObjectUtils.tryCast(expr != null ? expr.getParent() : null, ErlangArgumentList.class);
+      if (expr == null || args == null) return false;
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunctionCallExpression call = ObjectUtils.tryCast(args.getParent(), ErlangFunctionCallExpression.class);
+      ErlangFunction function = resolveToFunction(call);
+      if (function == null) return false;
+
+      for (ErlangFunctionClause clause : function.getFunctionClauseList()) {
+        if (elementSuits(clause, position)) return true;
+      }
+      return false;
+    }
+
+    private boolean elementSuits(@NotNull ErlangFunctionClause clause, int position) {
+      List<ErlangArgumentDefinition> definitions = clause.getArgumentDefinitionList().getArgumentDefinitionList();
+      ErlangArgumentDefinition definition = definitions.size() > position ? definitions.get(position) : null;
+      ErlangMaxExpression maxExpression = PsiTreeUtil.getChildOfType(definition, ErlangMaxExpression.class);
+      ErlangQVar var = PsiTreeUtil.getChildOfType(maxExpression, ErlangQVar.class);
```

Comment:
Do not abuse ```PsiTreeUtil.getChildOfType``` - you have pretty much all you need in your psi. ArgumentDefinition has a method to get it's expression, and MaxExpression has a method to get a variable - use them.